In [3]:
from sqlalchemy import create_engine
import pandas as pd

In [2]:
engine = create_engine("postgresql://train:Ankara06@localhost:5433/traindb")

# Read with table name

In [5]:
ratings = pd.read_sql_table('ratings', con=engine)

In [6]:
ratings.head()

user_id  item_id  rating  rating_time
0      196      242       3    881250949
1      186      302       3    891717742
2       22      377       1    878887116
3      244       51       2    880606923
4      166      346       1    886397596

# Read with query

In [7]:
ratings = pd.read_sql_query('select * from ratings', con=engine)

In [8]:
ratings.head()

user_id  item_id  rating  rating_time
0      196      242       3    881250949
1      186      302       3    891717742
2       22      377       1    878887116
3      244       51       2    880606923
4      166      346       1    886397596

# Write dataframe to database

In [10]:
ratings.to_sql('ratings_pandas', con=engine, index=False, if_exists='replace')

990

    select * from ratings_pandas limit 5;

    user_id|item_id|rating|rating_time|
    -------+-------+------+-----------+
        196|    242|     3|  881250949|
        186|    302|     3|  891717742|
         22|    377|     1|  878887116|
        244|     51|     2|  880606923|
        166|    346|     1|  886397596|